lib/validate.js

```
@@ -707,13 +708,19 @@ const _validateMainRelativePath = indexMetadata => {
   )
 }
 
-const _validateMainExtension = indexMetadata => {
+const _validateMainExtension = async (indexMetadata, key, p2pcommonsDir) => {
   function getExtension (filename) {
     var ext = extname(filename || '').split('.')
     return ext[ext.length - 1]
   }
 
   const ext = getExtension(indexMetadata.p2pcommons.main)
+  const mainPath = join(p2pcommonsDir, key, indexMetadata.p2pcommons.main)
+
+  let mainBuffer = null
+  try {
+    mainBuffer = await readFile(mainPath)
+  } catch (_) {}
```

what's the reason for ignoring errors here?

---

This code is part of `_validateMainExtension` function which is part of `validateMain` function. The `_validateMainExtension` runs after `_validateMainExists`, a function that checks for the existence of the main file. So we can assume that a main file should exist up to this point.

I think I wanted to be protected if the function run outside `validateMain` and/or something weird happens with readFile fn. This is connected to the fact that we are adding this double check like the extension check and the content one. If there is no `mainBuffer` for some reason, then the content check wont happen and we will end up with two possibly different validations on the extensions of the main file.  So the absence of the mainBuffer does not break things. Unless we strongly wanted to have the content validation.

---

I put some seconds thoughts on this and I think it's safe to remove that try/catch cage. Release the readFile! :)

---

I think we kind of have to do content validation, if "it's plain text" makes it valid. Good point about _validateMainExtension being callable outside of the validateMain function, I assumed it would be private because it starts the an underscore.